### DISCUSSION TITLE
Prediction protein structure using ESMFOLD2 - 'ESM2' object has no attribute 'infer'

### DISCUSSION THREAD

**Discussion by heliogabale** [facebookresearch/esm#587] [Q&A]
Hello
I am trying to use the pretrained esm2_t36_3B_UR50D() and esm2_t48_15B_UR50D() to predict protein structure.
I am taking my inspiration from the code for esmfold_v1()
model, alphabet = esm.pretrained.esm2_t36_3B_UR50D()

with torch.no_grad():
    output = model.infer(sequence)

I am getting 
AttributeError: 'ESM2' object has no attribute 'infer'
Thanks

**Comment by Ming-Qin-tech** [comment]
same question

**Comment by ebetica** [comment]
You should use the ESMFold model, ESM2 does't predict structure `model = esm.pretrained.esmfold_v1()`
